src/osdc/Objecter.h

```diff
@@ -1133,7 +1133,7 @@ class Objecter : public md_config_obs_t, public Dispatcher {
 
   PerfCounters *logger;
 
-  uint64_t tick_event;
+  atomic_t tick_event;
```

Comment:
All the accesses to tick_event are done inside Objecter::rwlock, apart from the assertion at the end of ::shutdown.  I think that assertion is just completely bogus, because it's not synchronised with respect to anything.  Rather than making this atomic, I think it makes more sense to get rid of the assertion.

It *used* to be an assertion that tick_event was zeroed after waiting for timer to shut down (when timer was a SafeTimer), but it looks like now that shutdown happens inside ~Objecter (in ~timer).  It might have been nice to give timer a shutdown method too: it feels a bit odd to me to leave threads running after shutdown and only join them in a destructor (but I'm not claiming it's actually functioning incorrectly in this instance).